pkg/controller/utils/elasticsearch.go

```diff
@@ -62,3 +131,261 @@ func GetElasticsearchClusterConfig(ctx context.Context, cli client.Client) (*ren
 
 	return render.NewElasticsearchClusterConfigFromConfigMap(configMap)
 }
+
+type ElasticClient interface {
+	NewElasticsearchClient(client.Client, context.Context) error
+	SetElasticsearchIlmPolicies(context.Context, *operatorv1.LogStorage, int64) error
+	GetElasticsearchClient() *elastic.Client
+}
+
+type EsClient struct {
+	client *elastic.Client
+}
+
+func (es *EsClient) NewElasticsearchClient(client client.Client, ctx context.Context) error {
+	user, password, root, err := getClientCredentials(client, ctx)
+	if err != nil {
+		return err
+	}
+
+	h := &http.Client{
+		Transport: &http.Transport{TLSClientConfig: &tls.Config{RootCAs: root}},
+	}
+
+	esClient, err := NewElastic(user, password, render.ElasticsearchHTTPSEndpoint, h)
+	if err != nil {
+		return err
+	}
+
+	es.client = esClient
+	return nil
+}
+
+// SetElasticsearchIlmPolicies creates ILM policies for each timeseries based index using the retention period and storage size in LogStorage
+func (es *EsClient) SetElasticsearchIlmPolicies(ctx context.Context, ls *operatorv1.LogStorage, totalEsStorage int64) error {
+	for _, v := range IndexDiskMapping {
```

Comment:
I was thinking if you would change this method to first call a method to create a list of policies and then a method to apply the policies, your code would become more readable. This struct IndexDiskMapping could disappear and be a variable inside of the method that created the policies